Delete `--incompatible_remote_use_new_exit_code_for_lost_inputs`

fmeum opened the pull request:
The flag has been flipped in Bazel 7 and constitutes a very minor change that nonetheless complicates remote execution exception handling.

tjgq reviewed (changes requested):
Do we have to keep it around for another year? I'm not sure if we have an explicit policy around this, but I'm pretty sure we've flipped incompatible flags in major release X and deleted them in major release X+1 in the past.

fmeum commented:
> Do we have to keep it around for another year? I'm not sure if we have an explicit policy around this, but I'm pretty sure we've flipped incompatible flags in major release X and deleted them in major release X+1 in the past.

Turns out it has been flipped in Bazel 7. I updated the PR to get rid of it entirely.
